applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+	struct cloud_data_battery bat_buf = {0};
+
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(-ENODATA, ret);
+	TEST_ASSERT_EQUAL(NULL, codec.buf);
+}
+
+void test_enc_batch_data_single_battery(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {0};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {
+		.bat = 50,
+		.bat_ts = 1563968747123,
+		.queued = true,
+	};
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, strncmp(BAT_BATCH_EXAMPLE, codec.buf, strlen(BAT_BATCH_EXAMPLE)));
+	TEST_ASSERT_FALSE(bat_buf.queued);
+}
+
+void test_enc_batch_data_single_battery_too_big(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {0};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {
+		.bat = INT16_MAX,
+		.bat_ts = 1563968747123,
+		.queued = true,
+	};
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(-ENOMEM, ret);
+	TEST_ASSERT_TRUE(bat_buf.queued);
+}
+
+void test_enc_batch_data_gnss(void)
+{
+	struct cloud_data_gnss gnss_buf = {
+		.format = CLOUD_CODEC_GNSS_FORMAT_NMEA,
+		.queued = true,
+		.gnss_ts = 1563968747123,
+		.nmea = "$GPGGA,"
+			"141856.00,6326.27336,N,01028.08721,E,1,09,0.81,34.7,M,39.8,M,,*64\n",
+	};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {0};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {0};
+
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, strncmp(GNSS_BATCH_EXAMPLE, codec.buf, strlen(GNSS_BATCH_EXAMPLE)));
+	TEST_ASSERT_FALSE(gnss_buf.queued);
+}
+
+void test_enc_batch_data_modem_static(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {
```

Comment:
Can you make this struct file global and declare it close to MODEM_STATIC_BATCH_EXAMPLE ? That we we know they are related and the test case also becomes small. (I think there are similar occurences in other tests too)

Reply:
done